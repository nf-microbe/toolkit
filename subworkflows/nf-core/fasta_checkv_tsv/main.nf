// Import modules
include { CHECKV_DOWNLOADDATABASE   } from '../../../modules/nf-core/checkv/downloaddatabase'
include { CHECKV_ENDTOEND           } from '../../../modules/nf-core/checkv/endtoend'
include { CHECKV_GENBANKHITS        } from '../../../modules/nf-core/checkv/genbankhits'

workflow FASTA_CHECKV_TSV {
    take:
    virus_fasta_gz  // channel: [ [ meta ], fasta.gz ]
    checkv_db       // path: checkv_db
    ncbi_info       // path: ncbi_info

    main:
    ch_versions = Channel.empty()

    // if checkv DB exists, skip CHECKV_DOWNLOADDATABASE
    if (checkv_db){
        ch_checkv_db = checkv_db
    } else {
        //
        // MODULE: download standard CheckV database
        //
        CHECKV_DOWNLOADDATABASE(
        )
        ch_checkv_db    = CHECKV_DOWNLOADDATABASE.out.checkv_db
        ch_versions     = ch_versions.mix(CHECKV_DOWNLOADDATABASE.out.versions)
    }

    //
    // MODULE: Run CheckV end to end
    //
    CHECKV_ENDTOEND(
        virus_fasta_gz,
        ch_checkv_db.first()
    )
    ch_versions = ch_versions.mix(CHECKV_ENDTOEND.out.versions)

    //
    // MODULE: Identify CheckV Genbank hits
    //
    CHECKV_GENBANKHITS(
        CHECKV_ENDTOEND.out.aai,
        ncbi_info,
        ch_checkv_db.first()
    )
    ch_versions     = ch_versions.mix(CHECKV_GENBANKHITS.out.versions)

    emit:
    quality_summary_tsv = CHECKV_ENDTOEND.out.quality_summary   // channel: [ [ meta ], quality_summary.tsv ]
    contamination_tsv   = CHECKV_ENDTOEND.out.contamination     // channel: [ [ meta ], contamination.tsv ]
    completeness_tsv    = CHECKV_ENDTOEND.out.completeness      // channel: [ [ meta ], completeness.tsv ]
    genbank_hits_tsv    = CHECKV_GENBANKHITS.out.genbank_hits   // path: [ [ meta ], genbank_hits.tsv ]
    versions            = ch_versions                           // path: [ versions.yml ]
}
