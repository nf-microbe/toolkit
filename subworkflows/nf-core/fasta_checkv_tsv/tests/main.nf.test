nextflow_workflow {

    name "Test Subworkflow FASTA_CHECKV_TSV"
    script "../main.nf"
    workflow "FASTA_CHECKV_TSV"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/fasta_checkv_tsv"
    tag "checkv"
    tag "checkv/downloaddatabase"
    tag "checkv/endtoend"
    tag "checkv/genbankhits"

    test("sarscov2 - genome - fasta") {
        when {
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id:'test1', single_end:false ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fasta/contigs.fasta', checkIfExists: true),
                    ],
                    [
                        [ id:'test2', single_end:false ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta.gz', checkIfExists: true),
                    ]
                )
                input[1] = null
                input[2] = "${projectDir}/assets/db/ncbi_info.tsv"
                """
            }
        }
        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.quality_summary_tsv,
                    workflow.out.contamination_tsv,
                    workflow.out.complete_genomes,
                    workflow.out.viruses_fasta_gz,
                    workflow.out.versions
                    ).match()
                }
            )
        }
    }
}
