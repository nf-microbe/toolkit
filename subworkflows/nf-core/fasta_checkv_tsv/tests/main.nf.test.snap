{
    "sarscov2 - genome - fasta": {
        "content": [
            [
                [
                    {
                        "id": "test1",
                        "single_end": false
                    },
                    "test1_quality_summary.tsv:md5,f7aea0429fd38068bb26be4e16dafd6d"
                ],
                [
                    {
                        "id": "test2",
                        "single_end": false
                    },
                    "test2_quality_summary.tsv:md5,7378f3a63fa5d7e4023f96aa12f0de1a"
                ]
            ],
            [
                [
                    {
                        "id": "test1",
                        "single_end": false
                    },
                    "test1_contamination.tsv:md5,c4072dbfbb3f876baf12f15d65c0e7db"
                ],
                [
                    {
                        "id": "test2",
                        "single_end": false
                    },
                    "test2_contamination.tsv:md5,ee7d35d11b98659798339fedb8e91bd6"
                ]
            ],
            null,
            null,
            [
                "versions.yml:md5,632ef17fe45cea27f0c4fbaab94586d0",
                "versions.yml:md5,632ef17fe45cea27f0c4fbaab94586d0",
                "versions.yml:md5,6802648b7424cba8470d94ec005a951d",
                "versions.yml:md5,bdef4ba3c5474d1d2dab482b3ae060b0",
                "versions.yml:md5,bdef4ba3c5474d1d2dab482b3ae060b0"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-10T07:33:27.452917757"
    }
}